Do not include `from` information when its not valid

stieg opened the pull request:
If you include the information then when the _execModule routine loads the
module without a from context then it will incorrectly setup a circular
dependency by declaring the parent is itself. By checking for a module name
and not including that information when passed in, the issue is avoided.

<!-- Thanks for submitting a pull request! Please provide enough information so that others can review your pull request. The two fields below are mandatory. -->

<!-- Please remember to update CHANGELOG.md in the root of the project if you have not done so. -->

## Summary

<!-- Explain the **motivation** for making this change. What existing problem does the pull request solve? -->
Fixes #5235 
## Test plan

<!-- Demonstrate the code is solid. Example: The exact commands you ran and their output, screenshots / videos if the pull request changes UI. -->
Before I go through and finish this up I want someone with more knowledge of this project to review this (arguably trivial) change and ensure it makes sense.  I have a headache from trying to figure all this out and AFAICT this is the best solution to solve the issue.

That said... its simple enough to load a mock test and trace up to the parent module and ensure it is null.

SimenB commented:
Thanks! Mind including the integration test from #5241 as well?

SimenB commented:
FWIW, I think this change makes sense 🙂 

If you rebase on master, it will fix an issue with CI which will crop up when you fix the flow errors currently reported

stieg commented:
@SimenB Hey so am confused what you mean by the integration test from #5241.  I have rebased onto master and am running things locally before I clobber and update the PR.  Can you help me out?

SimenB commented:
There's an integration test added in #5241. Could you copy it over and include it in this PR?

stieg commented:
Right... rebased on master; included those integration tests at your request, fixed linter and babel errors... only a couple of failures locally but they appear unrelated.  Over to you @SimenB

SimenB commented:
Local failures are probably missing mercurial (hg) installation, seeing as CI passed :)
